relay-server/src/actors/store.rs

```diff
@@ -1025,19 +1099,50 @@ struct CheckInKafkaMessage {
     retention_days: u16,
 }
 
+/// Schema inspired by but not equal to the sentry <-> snuba schema.
+/// See https://github.com/getsentry/sentry-kafka-schemas/blob/main/schemas/snuba-spans.v1.schema.json
+///
+/// Most notably, this schema does not contain the `group_raw`, which must be computed on sentry-side.
 #[derive(Debug, Serialize)]
 struct SpanKafkaMessage {
-    /// Raw span data. See [`relay_event_schema::protocol::Span`] for schema.
-    span: serde_json::Value,
-    /// Time at which the span was received by Relay.
+    /// Version of this message's format. Tells the consumer how to parse this message.
+    version: u16,
+    /// Time at which the event was received by Relay. Not to be confused with `start_timestamp_ms`.
     start_time: u64,
-    /// The project id for the current span.
+    /// The ID of the transaction event associated to this span, if any.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    event_id: Option<EventId>,
+    /// The numeric ID of the organization.
+    organization_id: u64,
+    /// The numeric ID of the project.
     project_id: ProjectId,
-    /// The event id for the current span.
+    /// The ID of the trace that the current span belongs to.
+    trace_id: String,
+    /// The unique ID of the current span.
+    span_id: String,
+    /// The ID of the current span's parent, if any.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    parent_span_id: Option<String>,
+    /// A unique identifier for a segment within a trace (8 byte hexadecimal string).
     ///
-    /// Once spans are truly standalone, this field will be omitted.
+    /// For spans embedded in transactions, the `segment_id` is the `span_id` of the containing
+    /// transaction.
     #[serde(skip_serializing_if = "Option::is_none")]
-    event_id: Option<EventId>,
+    segment_id: Option<String>,
+    /// Whether or not the current span is the root of the segment.
+    is_segment: bool,
+    /// Timestamp when the span started in milliseconds.
+    start_timestamp_ms: u64,
+    /// Total duration of the span in milliseconds.
+    duration_ms: u64,
+    /// The amount of time spent in this span, excluding its immediate child spans.
+    exclusive_time_ms: u64,
+    retention_days: u16,
```

Comment:
@phacops the `retention_days` field in combination with `organization_id` should make the [db lookup in the sentry consumer](https://github.com/getsentry/sentry/blob/808f610f97f78a5ad5df2381efecad642c368f63/src/sentry/spans/consumers/process/factory.py#L58-L60) unnecessary, which will hopefully save some resources.